R/status.R

```diff
@@ -1,10 +1,65 @@
 
 the$status_running <- FALSE
 
-#' Report differences between lockfile and project library
+#' Report inconsistencies between lockfile, library, and dependencies
 #'
-#' Report differences between the project's lockfile and the current state of
-#' the project's library (if any).
+#' `status()` reports issues caused by inconsistencies across the project
+#' lockfile, library, and [dependencies()]. In general, you should strive to
+#' ensure that `status()` reports no issues, as this maximises your chances of
+#' successfully `restore()`ing the project in the future or on another machine.
+#'
+#' See the headings below for specific advice on resolving any issues
+#' revealed by `status()`.
+#'
+#' # Missing packages
+#'
+#' First, ensure that all packages used by the project are installed. This is
+#' important to do first because if any packages are missing we can't tell for
+#' sure that a package isn't used; it might be a dependency that we don’t know
+#' about.
+#'
+#' There are four possibilities for an uninstalled package:
+#'
+#' * If it’s used and recorded, call `renv::restore()` to install the version
+#'   specified in the lockfile.
+#' * If it’s used and not recorded, call `renv::install()` to install it
+#'   from CRAN or elsewhere.
+#' * If it’s not used and recorded, call `renv::snapshot()` to
+#'   remove it from the lockfile.
+#' * If it's not used and not recorded, there's nothing to do. This the most
+#'   common state because you only use a small fraction of all available
+#'   packages in any one project.
+#'
+#' # Lockfile vs `dependencies()`
+#'
+#' Next we need to ensure that packages are recorded in the lockfile if and
+#' only if they are used by the project. Fixing issues of this nature only
+#' requires calling  `snapshot()` because there are four possibilities for
+#' a package:
+#'
+#' * If it’s used and recorded, it’s ok.
+#' * If it’s used and not recorded, call `renv::snapshot()` to add it to the
+#'   lockfile.
+#' * If it’s not used but is recorded, call `renv::snapshot()` to remove
+#'   it from the lockfile.
+#' * If it’s not used and not recorded, it’s also ok, as it may be a
+#'   development dependency.
+#'
+#' # Out-of-sync sources
+#'
+#' The final issue to resolve is any inconsistencies between the version of
+#' the package recorded in the lockfile and the version installed in your
+#' library. To fix these issues you'll need to either call `renv::restore()`
+#' or `renv::snapshot()`:
+#'
+#' * Call `renv::snapshot()` if your project code is working. This implies that
+#'   the library is correct and you need to update your lockfile.
+#' * Call `renv::restore()` if your project code isn't working. This probably
+#'   implies that you have the wrong package versions installed and you need
+#'   to restore from known good state in the lockfile.
+#'
+#' If you're not sure which case applies, it's generally safer to call
+#' `renv::snapshot()`
```

Comment:
We might also recommend using `git` + `renv::history()` or `renv::revert()` for finding and using historical versions of lockfiles.